ingestr/src/sources.py

```diff
@@ -2392,3 +2392,35 @@ def dlt_source(self, uri: str, table: str, **kwargs):
             verify_certs=verify_certs,
             incremental=incremental,
         ).with_resources(table)
+
+
+class AttioSource:
+    def handles_incrementality(self) -> bool:
+        return True
+
+    def dlt_source(self, uri: str, table: str, **kwargs):
+        parsed_uri = urlparse(uri)
+        query_params = parse_qs(parsed_uri.query)
+        api_key = query_params.get("api_key")
+
+        if api_key is None:
+            raise MissingValueError("api_key", "Attio")
+
+        parts = table.replace(" ", "").split(":")
+        table_name = parts[0]
+        params = parts[1:]
+
+        if table_name not in [
```

Comment:
There's a trick to validation resources that's more robust than this.

Once a source is constructed, you can reference it's `resources` field. This means that you never have to update this list, even if you add or remove resources in the future.

```py
src = attio_source(api_key=api_key[0], params=params)
if table_name not in src.resources:
  raise ...
```

Reply:
here's an example: https://github.com/bruin-data/ingestr/blob/main/ingestr/src/sources.py#L1700